### PR TITLE
trace: add traces defines to kconfig.

### PR DESCRIPTION
**Pull request by bkokoszx**
Add TRACE, TRACEV, TRACEE, TRACEM to Kconfig.

Signed-off-by: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>

### CODE REVIEW

File: src/include/sof/trace.h [hunk 1]
```diff
@@ -212,7 +206,7 @@ void trace_flush(void);
 void trace_off(void);
 void trace_init(struct sof *sof);
 
-#if TRACE
+#if defined(CONFIG_TRACE)
```
Comment (jajanusz) [inline src/include/sof/trace.h]: Please use `#if CONFIG_*`, ifdefs break when value evaluate to false

Comment (bkokoszx) [reply src/include/sof/trace.h]: I've changed.

**Comment by bkokoszx** [comment]
@jajanusz @lgirdwood 
I've removed trace core logic from build when CONFIG_TRACE is not set by Kconfig in additional commit. I'm not sure whether FW should returns error when driver tries to enable dma traces in FW built without CONFIG_TRACE? Should trace ipc be ignored (```ipc_glb_debug_message()``` returns 0)?

File: src/lib/CMakeLists.txt [hunk 2]
```diff
@@ -8,16 +8,19 @@ add_local_sources(sof
...
 	pm_runtime.c
 	clk.c
 	dma.c
 	dai.c
 	panic.c
 	wait.c
 )
+
+if (CONFIG_TRACE)
+add_local_sources(sof dma-trace.c
```
Comment (jajanusz) [inline src/lib/CMakeLists.txt]: Please format more like:
```
add_local_sources(<target>
	<src1>
	...
	<srcN>
)
```

**Comment by jajanusz** [comment]
> @jajanusz @lgirdwood
> I've removed trace core logic from build when CONFIG_TRACE is not set by Kconfig in additional commit. I'm not sure whether FW should returns error when driver tries to enable dma traces in FW built without CONFIG_TRACE? Should trace ipc be ignored (`ipc_glb_debug_message()` returns 0)?

I think it's always better to return error (instead of ignoring) when there is some misuse

File: src/lib/CMakeLists.txt (hunk 2 above)
Comment (bkokoszx) [reply src/lib/CMakeLists.txt]: It's done.

File: src/ipc/CMakeLists.txt [hunk 3]
```diff
@@ -10,5 +10,9 @@ endif()
 add_local_sources(sof
 	ipc.c
 	handler.c
-	dma-copy.c
 )
+
+if (CONFIG_TRACE)
+add_local_sources(sof
```
Comment (jajanusz) [inline src/ipc/CMakeLists.txt]: Please use indention in CMake just like in C (increase indent in ifs, functions etc)

Comment (bkokoszx) [reply src/ipc/CMakeLists.txt]: Done.

**Comment by bardliao** [comment]
> I think it's always better to return error (instead of ignoring) when there is some misuse

Yes, and kernel will know trace is not supported by FW.

**Comment by bkokoszx** [comment]
@bardliao 
Ok, so in that case FW returns -EINVAL error.

**Comment by bkokoszx** [comment]
@RanderWang 
I've added ```#define SRAM_TRACE_SIZE 0x0``` when CONFIG_TRACE is not set.

**Comment by RanderWang** [comment]
> @RanderWang
> I've added `#define SRAM_TRACE_SIZE 0x0` when CONFIG_TRACE is not set.

It is MAILBOX_TRACE_SIZE on BYT and HSW, Thanks.

**Comment by bkokoszx** [comment]
@RanderWang 
Sorry, I've missed that. Updated.

**Pull request merged**